experiments/react-native-uifabric/src/components/RadioGroup/RadioButton.tsx

```diff
@@ -0,0 +1,140 @@
+
+export const RadioButton = compose<IRadioButtonType>({
+  displayName: radioButtonName,
+
+  usePrepareProps: (userProps: IRadioButtonProps, useStyling: IUseComposeStyling<IRadioButtonType>) => {
+    const { content, buttonKey, disabled, ariaLabel, ...rest } = userProps;
+
+    // Grabs the context information from RadioGroup (currently selected button and client's onChange callback)
+    const info = React.useContext(RadioGroupContext);
+
+    const pressable = useAsPressable({
+      ...rest
+    });
+
+    const state: IRadioButtonState = {
+      ...pressable.state,
+      selected: info.selectedKey == userProps.buttonKey ? true : false,
+      disabled: disabled ? disabled : false
+    };
+
+    // Grab the styling information from the userProps, referencing the state as well as the props.
+    const styleProps = useStyling(userProps, (override: string) => state[override] || userProps[override]);
+
+    // This function is called every time a RadioButton gains focus. It does two things:
+    // 1) Calls pressable's onFocus in order to keep track of our state's focus variable. It is dependent on pressable's
+    //    focus variable. Without this, it wouldn't stay updated because we're overriding it's onFocus below for the rootProps.
+    // 2) Selects the currently focused button by calling the RadioGroup's callback function.
+    const onFocusChange = (ev: NativeSyntheticEvent<{}>) => {
+      pressable.props.onFocus && pressable.props.onFocus(ev);
+      // This check is necessary because this func gets called even when a button loses focus (not sure why?) which then calls the client's onChange multiple times
+      if (!state.selected) info.onButtonSelect(buttonKey);
+    };
+
+    const accessibilityTraits = {
+      accessibilityRole: 'radio',
+      accessibilityLabel: ariaLabel ? ariaLabel : content,
+      // Why are we able to do this? The radio button that's focused is always going to be the one selected. Therefore, unless it's disabled,
+      // we can always set state=selected for the narrator, because every time the narrator is focused on a button, that's the one selected
+      accessibilityStates: state.disabled ? ['disabled'] : ['selected'],
+      accessibilityActions: [{ name: 'Select', label: 'Select a RadioButton' }],
+      onAccessibilityAction: (event: { nativeEvent: { actionName: any } }) => {
+        switch (event.nativeEvent.actionName) {
+          case 'Select':
+            info.onButtonSelect(buttonKey);
+        }
+      }
+    };
+
+    //When the option is disabled, there's no built-in disable functionality for React-Native. So here's the logic I implemented for it.
+    const rootProps = state.disabled
+      ? { rest, ...accessibilityTraits }
+      : {
+          ...pressable.props,
+          ...accessibilityTraits,
+          rest,
+          onFocus: onFocusChange
+        };
+    const buttonProps = state.disabled ? { rest, style: { borderColor: 'grey' } } : rest;
```

Comment:
So any calculations of styles should come from the style functions that are passed to the `styles` prop below, so that they can be cached. We should avoid constructing style objects in this function.

Reply:
This comment applies to all other places in this function and the RadioGroup control.

Reply:
This can be done in settings using _overrides, no?

Reply:
Replaced this using _overrides in my newest iteration